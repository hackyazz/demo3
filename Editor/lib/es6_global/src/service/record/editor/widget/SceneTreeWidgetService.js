


function getWidget(param) {
  return /* SceneTree */0;
}

function isWidget(startWidget) {
  if (startWidget !== undefined) {
    return startWidget === /* SceneTree */0;
  } else {
    return false;
  }
}

export {
  getWidget ,
  isWidget ,
  
}
/* No side effect */
