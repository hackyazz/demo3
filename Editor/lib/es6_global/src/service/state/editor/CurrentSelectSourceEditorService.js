


function getCurrentSelectSource(editorState) {
  return editorState[/* currentSelectSource */12];
}

function setCurrentSelectSource(currentSelectSource, editorState) {
  return /* record */[
          /* settingRecord */editorState[/* settingRecord */0],
          /* sceneTreeRecord */editorState[/* sceneTreeRecord */1],
          /* assetRecord */editorState[/* assetRecord */2],
          /* sceneViewRecord */editorState[/* sceneViewRecord */3],
          /* gameViewRecord */editorState[/* gameViewRecord */4],
          /* eventRecord */editorState[/* eventRecord */5],
          /* imguiRecord */editorState[/* imguiRecord */6],
          /* inspectorRecord */editorState[/* inspectorRecord */7],
          /* consoleRecord */editorState[/* consoleRecord */8],
          /* transformRecord */editorState[/* transformRecord */9],
          /* pickingRecord */editorState[/* pickingRecord */10],
          /* currentDragSource */editorState[/* currentDragSource */11],
          /* currentSelectSource */currentSelectSource,
          /* loopId */editorState[/* loopId */13],
          /* languageType */editorState[/* languageType */14]
        ];
}

function clearCurrentSelectSource(editorState) {
  return /* record */[
          /* settingRecord */editorState[/* settingRecord */0],
          /* sceneTreeRecord */editorState[/* sceneTreeRecord */1],
          /* assetRecord */editorState[/* assetRecord */2],
          /* sceneViewRecord */editorState[/* sceneViewRecord */3],
          /* gameViewRecord */editorState[/* gameViewRecord */4],
          /* eventRecord */editorState[/* eventRecord */5],
          /* imguiRecord */editorState[/* imguiRecord */6],
          /* inspectorRecord */editorState[/* inspectorRecord */7],
          /* consoleRecord */editorState[/* consoleRecord */8],
          /* transformRecord */editorState[/* transformRecord */9],
          /* pickingRecord */editorState[/* pickingRecord */10],
          /* currentDragSource */editorState[/* currentDragSource */11],
          /* currentSelectSource */undefined,
          /* loopId */editorState[/* loopId */13],
          /* languageType */editorState[/* languageType */14]
        ];
}

export {
  getCurrentSelectSource ,
  setCurrentSelectSource ,
  clearCurrentSelectSource ,
  
}
/* No side effect */
