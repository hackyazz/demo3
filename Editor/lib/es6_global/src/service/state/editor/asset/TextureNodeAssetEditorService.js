

import * as List from "../../../../../../../node_modules/bs-platform/lib/es6/list.js";
import * as Js_option from "../../../../../../../node_modules/bs-platform/lib/es6/js_option.js";
import * as ArrayService$WonderEditor from "../../../atom/ArrayService.js";
import * as NodeAssetEditorService$WonderEditor from "./NodeAssetEditorService.js";
import * as TreeAssetEditorService$WonderEditor from "./TreeAssetEditorService.js";
import * as IterateTreeAssetService$WonderEditor from "../../../record/editor/asset/IterateTreeAssetService.js";
import * as TextureNodeAssetService$WonderEditor from "../../../record/editor/asset/TextureNodeAssetService.js";
import * as IterateTreeAssetEditorService$WonderEditor from "./IterateTreeAssetEditorService.js";
import * as OperateTreeAssetEditorService$WonderEditor from "./OperateTreeAssetEditorService.js";

function setNodeData(nodeId, nodeData, editorState) {
  return NodeAssetEditorService$WonderEditor.setNodeData(nodeId, nodeData, TextureNodeAssetService$WonderEditor.buildNodeByNodeData, editorState);
}

function unsafeGetNodeData(nodeId, editorState) {
  return TextureNodeAssetService$WonderEditor.getNodeData(OperateTreeAssetEditorService$WonderEditor.unsafeFindNodeById(nodeId, editorState));
}

var addTextureNodeToAssetTree = NodeAssetEditorService$WonderEditor.addNodeToAssetTree;

function doesAnyTextureUseImage(targetImageDataIndex, editorState) {
  return List.length(IterateTreeAssetService$WonderEditor.filter(TreeAssetEditorService$WonderEditor.unsafeGetTree(editorState), /* [] */0, (function (node, acc) {
                    return /* :: */[
                            node,
                            acc
                          ];
                  }), (function (node) {
                    var match = TextureNodeAssetService$WonderEditor.getNodeData(node);
                    return match[/* imageDataIndex */1] === targetImageDataIndex;
                  }), undefined, undefined, undefined, /* () */0)) > 0;
}

function getDataByTextureComponent(targetTextureComponent, editorState) {
  return Js_option.map(TextureNodeAssetService$WonderEditor.getNodeData, IterateTreeAssetService$WonderEditor.findOne(TreeAssetEditorService$WonderEditor.unsafeGetTree(editorState), (function (node) {
                    var match = TextureNodeAssetService$WonderEditor.getNodeData(node);
                    return match[/* textureComponent */0] === targetTextureComponent;
                  }), undefined, undefined, undefined, /* () */0));
}

function findAllTextureNodes(editorState) {
  return IterateTreeAssetEditorService$WonderEditor.filter(editorState, /* array */[], ArrayService$WonderEditor.push, (function (node) {
                return true;
              }), undefined, undefined, undefined, /* () */0);
}

function getTextureComponents(editorState) {
  return findAllTextureNodes(editorState).map((function (node) {
                return TextureNodeAssetService$WonderEditor.getNodeData(node)[/* textureComponent */0];
              }));
}

export {
  setNodeData ,
  unsafeGetNodeData ,
  addTextureNodeToAssetTree ,
  doesAnyTextureUseImage ,
  getDataByTextureComponent ,
  findAllTextureNodes ,
  getTextureComponents ,
  
}
/* ArrayService-WonderEditor Not a pure module */
