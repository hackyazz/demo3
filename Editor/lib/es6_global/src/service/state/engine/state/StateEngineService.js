

import * as StateAPI$Wonderjs from "../../../../../../../node_modules/wonder.js/lib/es6_global/src/api/StateAPI.js";

var deepCopyForRestore = StateAPI$Wonderjs.deepCopyForRestore;

var createState = StateAPI$Wonderjs.createState;

var createStateData = StateAPI$Wonderjs.createStateData;

var getStateFromData = StateAPI$Wonderjs.getStateFromData;

var unsafeGetState = StateAPI$Wonderjs.unsafeGetState;

var setStateToData = StateAPI$Wonderjs.setStateToData;

var setState = StateAPI$Wonderjs.setState;

var setIsDebug = StateAPI$Wonderjs.setIsDebug;

export {
  deepCopyForRestore ,
  createState ,
  createStateData ,
  getStateFromData ,
  unsafeGetState ,
  setStateToData ,
  setState ,
  setIsDebug ,
  
}
/* StateAPI-Wonderjs Not a pure module */
