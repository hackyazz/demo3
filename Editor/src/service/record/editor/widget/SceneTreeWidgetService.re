open WidgetType;

let getWidget = () => SceneTree;

let isWidget = startWidget =>
  switch (startWidget) {
  | None => false
  | Some(startWidget) => startWidget === getWidget()
  };