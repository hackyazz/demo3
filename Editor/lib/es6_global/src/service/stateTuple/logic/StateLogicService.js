

import * as Curry from "../../../../../../node_modules/bs-platform/lib/es6/curry.js";
import * as StateEditorService$WonderEditor from "../../state/editor/StateEditorService.js";
import * as StateEngineService$WonderEditor from "../../state/engine/state/StateEngineService.js";
import * as DirectorEngineService$WonderEditor from "../../state/engine/DirectorEngineService.js";

function getEngineStateToGetData(handleFunc) {
  return Curry._1(handleFunc, StateEngineService$WonderEditor.unsafeGetState(/* () */0));
}

function getAndSetEngineState(handleFunc) {
  StateEngineService$WonderEditor.setState(Curry._1(handleFunc, StateEngineService$WonderEditor.unsafeGetState(/* () */0)));
  return /* () */0;
}

function refreshEngineState(engineState) {
  StateEngineService$WonderEditor.setState(DirectorEngineService$WonderEditor.loopBody(0, engineState));
  return /* () */0;
}

function refreshEngineStateAndReturnEngineState(engineState) {
  return DirectorEngineService$WonderEditor.loopBody(0, engineState);
}

function getAndRefreshEngineState(param) {
  StateEngineService$WonderEditor.setState(DirectorEngineService$WonderEditor.loopBody(0, StateEngineService$WonderEditor.unsafeGetState(/* () */0)));
  return /* () */0;
}

function getAndRefreshEngineStateWithFunc(handleFunc) {
  StateEngineService$WonderEditor.setState(DirectorEngineService$WonderEditor.loopBody(0, Curry._1(handleFunc, StateEngineService$WonderEditor.unsafeGetState(/* () */0))));
  return /* () */0;
}

function loopBodyWhenStop(engineState) {
  var match = StateEditorService$WonderEditor.getIsRun(/* () */0);
  if (match) {
    return engineState;
  } else {
    return DirectorEngineService$WonderEditor.loopBody(0, engineState);
  }
}

function getAndRefreshEngineStateWhenStop(param) {
  StateEngineService$WonderEditor.setState(loopBodyWhenStop(StateEngineService$WonderEditor.unsafeGetState(/* () */0)));
  return /* () */0;
}

function renderWhenStop(engineState) {
  var match = StateEditorService$WonderEditor.getIsRun(/* () */0);
  if (match) {
    return engineState;
  } else {
    return DirectorEngineService$WonderEditor.loopBody(0, engineState);
  }
}

function getEditorState(handleFunc) {
  return Curry._1(handleFunc, StateEditorService$WonderEditor.getState(/* () */0));
}

function getAndSetEditorState(handleFunc) {
  StateEditorService$WonderEditor.setState(Curry._1(handleFunc, StateEditorService$WonderEditor.getState(/* () */0)));
  return /* () */0;
}

function getStateToGetData(handleFunc) {
  return Curry._1(handleFunc, /* tuple */[
              StateEditorService$WonderEditor.getState(/* () */0),
              StateEngineService$WonderEditor.unsafeGetState(/* () */0)
            ]);
}

function getAndSetState(handleFunc) {
  var match = Curry._1(handleFunc, /* tuple */[
        StateEditorService$WonderEditor.getState(/* () */0),
        StateEngineService$WonderEditor.unsafeGetState(/* () */0)
      ]);
  StateEditorService$WonderEditor.setState(match[0]);
  StateEngineService$WonderEditor.setState(match[1]);
  return /* () */0;
}

function setState(param) {
  StateEditorService$WonderEditor.setState(param[0]);
  StateEngineService$WonderEditor.setState(param[1]);
  return /* () */0;
}

export {
  getEngineStateToGetData ,
  getAndSetEngineState ,
  refreshEngineState ,
  refreshEngineStateAndReturnEngineState ,
  getAndRefreshEngineState ,
  getAndRefreshEngineStateWithFunc ,
  loopBodyWhenStop ,
  getAndRefreshEngineStateWhenStop ,
  renderWhenStop ,
  getEditorState ,
  getAndSetEditorState ,
  getStateToGetData ,
  getAndSetState ,
  setState ,
  
}
/* StateEditorService-WonderEditor Not a pure module */
