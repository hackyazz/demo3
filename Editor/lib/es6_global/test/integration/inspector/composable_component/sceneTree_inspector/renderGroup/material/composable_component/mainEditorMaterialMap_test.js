

import * as Curry from "../../../../../../../../../../node_modules/bs-platform/lib/es6/curry.js";
import * as Sinon from "../../../../../../../../../../node_modules/wonder-bs-sinon/lib/es6_global/src/sinon.js";
import * as Sinon$1 from "sinon";
import * as Wonder_jest from "../../../../../../../../../../node_modules/wonder-bs-jest/lib/es6_global/src/wonder_jest.js";
import * as ReactTestRenderer from "react-test-renderer";
import * as WDBTool$WonderEditor from "../../../../../../../tool/WDBTool.js";
import * as LoadTool$WonderEditor from "../../../../../../asset/tool/LoadTool.js";
import * as TestTool$WonderEditor from "../../../../../../../tool/TestTool.js";
import * as DragWDBTool$WonderEditor from "../../../../../../tool/DragWDBTool.js";
import * as LoadWDBTool$WonderEditor from "../../../../../../tool/LoadWDBTool.js";
import * as InspectorTool$WonderEditor from "../../../../../../../tool/ui/InspectorTool.js";
import * as ReactTestTool$WonderEditor from "../../../../../../../tool/ReactTestTool.js";
import * as GameObjectTool$WonderEditor from "../../../../../../../tool/GameObjectTool.js";
import * as EventListenerTool$WonderEditor from "../../../../../../../unit/tool/EventListenerTool.js";
import * as StateLogicService$WonderEditor from "../../../../../../../../src/service/stateTuple/logic/StateLogicService.js";
import * as BuildComponentTool$WonderEditor from "../../../../../../../tool/BuildComponentTool.js";
import * as StateEngineService$WonderEditor from "../../../../../../../../src/service/state/engine/StateEngineService.js";
import * as MainEditorAssetTool$WonderEditor from "../../../../../../asset/tool/MainEditorAssetTool.js";
import * as MainEditorSceneTool$WonderEditor from "../../../../../../../tool/MainEditorSceneTool.js";
import * as AssetTreeInspectorTool$WonderEditor from "../../../../assetTree_inspector/tool/AssetTreeInspectorTool.js";
import * as SceneTreeWidgetService$WonderEditor from "../../../../../../../../src/service/record/editor/widget/SceneTreeWidgetService.js";
import * as MainEditorAssetTreeTool$WonderEditor from "../../../../../../asset/tool/MainEditorAssetTreeTool.js";
import * as MainEditorMaterialMapTool$WonderEditor from "../tool/MainEditorMaterialMapTool.js";
import * as NoWorkerJobConfigToolEngine$WonderEditor from "../../../../../../../tool/engine/NoWorkerJobConfigToolEngine.js";
import * as TextureNodeAssetEditorService$WonderEditor from "../../../../../../../../src/service/state/editor/asset/TextureNodeAssetEditorService.js";
import * as CurrentSelectSourceEditorService$WonderEditor from "../../../../../../../../src/service/state/editor/CurrentSelectSourceEditorService.js";

describe("test MainEditorMaterialMap", (function () {
        var sandbox = Sinon.getSandboxDefaultVal(/* () */0);
        var _prepare = function () {
          MainEditorSceneTool$WonderEditor.initState(sandbox, undefined, undefined, /* () */0);
          Curry._1(EventListenerTool$WonderEditor.stubGetElementByIdReturnFakeDom, Curry._1(EventListenerTool$WonderEditor.buildFakeDom, /* () */0));
          Curry._1(MainEditorAssetTool$WonderEditor.buildFakeFileReader, /* () */0);
          LoadTool$WonderEditor.buildFakeTextDecoder(LoadTool$WonderEditor.convertUint8ArrayToBuffer);
          LoadTool$WonderEditor.buildFakeURL(sandbox[0]);
          LoadTool$WonderEditor.buildFakeLoadImage();
          var partial_arg = SceneTreeWidgetService$WonderEditor.getWidget(/* () */0);
          return StateLogicService$WonderEditor.getAndSetEditorState((function (param) {
                        return CurrentSelectSourceEditorService$WonderEditor.setCurrentSelectSource(partial_arg, param);
                      }));
        };
        beforeEach((function () {
                sandbox[0] = Sinon$1.sandbox.create();
                return /* () */0;
              }));
        afterEach((function () {
                return Curry._1(Sinon.restoreSandbox, sandbox[0]);
              }));
        Wonder_jest.testPromise("load asset wdb;\n      drag wdb;\n      select wdb;\n\n      inspector->material map should show its map", (function () {
                _prepare(/* () */0);
                return DragWDBTool$WonderEditor.testDragWDB(sandbox, /* tuple */[
                            "CubeTextured",
                            WDBTool$WonderEditor.convertGLBToWDB("CubeTextured")
                          ], (function (_, _$1, _$2) {
                              GameObjectTool$WonderEditor.setCurrentSceneTreeNode(LoadWDBTool$WonderEditor.getCubeTexturedMeshGameObject(StateEngineService$WonderEditor.unsafeGetState(/* () */0)));
                              return Promise.resolve(ReactTestTool$WonderEditor.createSnapshotAndMatch(BuildComponentTool$WonderEditor.buildInspectorComponent(TestTool$WonderEditor.buildEmptyAppState(/* () */0), InspectorTool$WonderEditor.buildFakeAllShowComponentConfig(/* () */0))));
                            }));
              }));
        describe("test select texture group -> show order", (function () {
                beforeEach((function () {
                        sandbox[0] = Sinon$1.sandbox.create();
                        MainEditorSceneTool$WonderEditor.initStateWithJob(sandbox, NoWorkerJobConfigToolEngine$WonderEditor.buildNoWorkerJobConfig(undefined, undefined, undefined, undefined, undefined, /* () */0), undefined, undefined, undefined, /* () */0);
                        Curry._1(EventListenerTool$WonderEditor.stubGetElementByIdReturnFakeDom, Curry._1(EventListenerTool$WonderEditor.buildFakeDom, /* () */0));
                        return MainEditorSceneTool$WonderEditor.createDefaultScene(sandbox, MainEditorSceneTool$WonderEditor.setFirstCubeToBeCurrentSceneTreeNode);
                      }));
                afterEach((function () {
                        return Curry._1(Sinon.restoreSandbox, sandbox[0]);
                      }));
                return Wonder_jest.test("\n        order should be:\n        sort texture assets by firstname alphabetically\n        ", (function () {
                              var assetTreeData = Curry._1(MainEditorAssetTreeTool$WonderEditor.BuildAssetTree[/* Texture */2][/* buildTwoTextureAssetTree */1], /* () */0);
                              AssetTreeInspectorTool$WonderEditor.Rename[/* renameAssetTextureNode */1](undefined, undefined, Curry._1(MainEditorAssetTreeTool$WonderEditor.BuildAssetTree[/* Texture */2][/* getFirstTextureNodeId */2], assetTreeData), "BTexture", /* () */0);
                              AssetTreeInspectorTool$WonderEditor.Rename[/* renameAssetTextureNode */1](undefined, undefined, Curry._1(MainEditorAssetTreeTool$WonderEditor.BuildAssetTree[/* Texture */2][/* getSecondTextureNodeId */3], assetTreeData), "ATexture", /* () */0);
                              return ReactTestTool$WonderEditor.createSnapshotAndMatch(BuildComponentTool$WonderEditor.buildMaterialMap(undefined, undefined, true, undefined, /* () */0));
                            }));
              }));
        describe("fix bug", (function () {
                beforeEach((function () {
                        sandbox[0] = Sinon$1.sandbox.create();
                        MainEditorSceneTool$WonderEditor.initStateWithJob(sandbox, NoWorkerJobConfigToolEngine$WonderEditor.buildNoWorkerJobConfig(undefined, undefined, undefined, undefined, undefined, /* () */0), undefined, undefined, undefined, /* () */0);
                        Curry._1(EventListenerTool$WonderEditor.stubGetElementByIdReturnFakeDom, Curry._1(EventListenerTool$WonderEditor.buildFakeDom, /* () */0));
                        return MainEditorSceneTool$WonderEditor.createDefaultScene(sandbox, MainEditorSceneTool$WonderEditor.setFirstCubeToBeCurrentSceneTreeNode);
                      }));
                afterEach((function () {
                        return Curry._1(Sinon.restoreSandbox, sandbox[0]);
                      }));
                return Wonder_jest.test("test if has current texture component;\n         show texture group;\n        the current texture dom className should be select-item-active\n      ", (function () {
                              var assetTreeData = Curry._1(MainEditorAssetTreeTool$WonderEditor.BuildAssetTree[/* Texture */2][/* buildTwoTextureAssetTree */1], /* () */0);
                              var firstTextureNodeId = Curry._1(MainEditorAssetTreeTool$WonderEditor.BuildAssetTree[/* Texture */2][/* getFirstTextureNodeId */2], assetTreeData);
                              var match = StateLogicService$WonderEditor.getEditorState((function (param) {
                                      return TextureNodeAssetEditorService$WonderEditor.unsafeGetNodeData(firstTextureNodeId, param);
                                    }));
                              return ReactTestTool$WonderEditor.createSnapshotAndMatch(ReactTestRenderer.create(MainEditorMaterialMapTool$WonderEditor.renderTextureGroup(/* record */[
                                                  /* style */{
                                                    opacity: "1"
                                                  },
                                                  /* isShowTextureGroup */true,
                                                  /* currentTextureComponent */match[/* textureComponent */0]
                                                ], (function () {
                                                    return /* () */0;
                                                  }))));
                            }));
              }));
        return /* () */0;
      }));

export {
  
}
/*  Not a pure module */
