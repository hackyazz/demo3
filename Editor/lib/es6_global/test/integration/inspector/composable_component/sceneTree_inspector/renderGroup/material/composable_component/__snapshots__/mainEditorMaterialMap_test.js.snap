// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test MainEditorMaterialMap fix bug test if has current texture component;
         show texture group;
        the current texture dom className should be select-item-active
       1`] = `
<div
  className="select-component-content"
>
  <div
    className="select-component-item"
  >
    <div
      className="select-item-header"
    >
      Texture
    </div>
    <div
      className="select-item-imgBody"
    >
      <div
        className="imgBody-content"
      >
        <div
          className="select-item-imgContent select-item-active"
          onClick={[Function]}
        >
          <img
            className="imgContent-img"
            src="texture1img"
          />
          <div
            className="imgContent-text"
          >
            texture1
          </div>
        </div>
        <div
          className="select-item-imgContent"
          onClick={[Function]}
        >
          <img
            className="imgContent-img"
            src="texture2img"
          />
          <div
            className="imgContent-text"
          >
            texture2
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="select-component-bg"
    onClick={[Function]}
  />
</div>
`;

exports[`test MainEditorMaterialMap load asset wdb;
      drag wdb;
      select wdb;

      inspector->material map should show its map 1`] = `
<article
  className="wonder-inspector-component"
>
  <article
    className="wonder-inspector-sceneTree"
  >
    <div
      className="sceneTree-name"
    >
      <article
        className="inspector-item"
      >
        <div
          className="item-header"
        >
          <span
            className="component-label"
          >
            Name
          </span>
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="Mesh"
          />
        </div>
      </article>
    </div>
    <article
      className="componentBox-component"
    >
      <div
        className="header"
      >
        <div
          className="header-triangle"
          onClick={[Function]}
        >
          <span
            className="triangle-bottom"
          />
        </div>
        <div
          className="header-title"
        >
          Transform
        </div>
      </div>
      <article
        className="wonder-inspector-transform"
      >
        <article
          className="inspector-item wonder-three-float-input"
        >
          <div
            className="item-header"
          >
            Position
          </div>
          <div
            className="item-content"
          >
            <article
              className="inspector-item wonder-float-input"
            >
              <div
                className="item-header component-label"
                onMouseDown={[Function]}
                onMouseMove={[Function]}
                onMouseUp={[Function]}
              >
                X
              </div>
              <div
                className="item-content"
              >
                <input
                  className="input-component float-input"
                  onBlur={[Function]}
                  onChange={[Function]}
                  type="text"
                  value="0"
                />
              </div>
            </article>
            <article
              className="inspector-item wonder-float-input"
            >
              <div
                className="item-header component-label"
                onMouseDown={[Function]}
                onMouseMove={[Function]}
                onMouseUp={[Function]}
              >
                Y
              </div>
              <div
                className="item-content"
              >
                <input
                  className="input-component float-input"
                  onBlur={[Function]}
                  onChange={[Function]}
                  type="text"
                  value="0"
                />
              </div>
            </article>
            <article
              className="inspector-item wonder-float-input"
            >
              <div
                className="item-header component-label"
                onMouseDown={[Function]}
                onMouseMove={[Function]}
                onMouseUp={[Function]}
              >
                Z
              </div>
              <div
                className="item-content"
              >
                <input
                  className="input-component float-input"
                  onBlur={[Function]}
                  onChange={[Function]}
                  type="text"
                  value="0"
                />
              </div>
            </article>
          </div>
        </article>
        <article
          className="inspector-item wonder-three-float-input"
        >
          <div
            className="item-header"
          >
            Rotation
          </div>
          <div
            className="item-content"
          >
            <article
              className="inspector-item wonder-float-input"
            >
              <div
                className="item-header component-label"
                onMouseDown={[Function]}
                onMouseMove={[Function]}
                onMouseUp={[Function]}
              >
                X
              </div>
              <div
                className="item-content"
              >
                <input
                  className="input-component float-input"
                  onBlur={[Function]}
                  onChange={[Function]}
                  type="text"
                  value="0"
                />
              </div>
            </article>
            <article
              className="inspector-item wonder-float-input"
            >
              <div
                className="item-header component-label"
                onMouseDown={[Function]}
                onMouseMove={[Function]}
                onMouseUp={[Function]}
              >
                Y
              </div>
              <div
                className="item-content"
              >
                <input
                  className="input-component float-input"
                  onBlur={[Function]}
                  onChange={[Function]}
                  type="text"
                  value="0"
                />
              </div>
            </article>
            <article
              className="inspector-item wonder-float-input"
            >
              <div
                className="item-header component-label"
                onMouseDown={[Function]}
                onMouseMove={[Function]}
                onMouseUp={[Function]}
              >
                Z
              </div>
              <div
                className="item-content"
              >
                <input
                  className="input-component float-input"
                  onBlur={[Function]}
                  onChange={[Function]}
                  type="text"
                  value="0"
                />
              </div>
            </article>
          </div>
        </article>
        <article
          className="inspector-item wonder-three-float-input"
        >
          <div
            className="item-header"
          >
            Scale
          </div>
          <div
            className="item-content"
          >
            <article
              className="inspector-item wonder-float-input"
            >
              <div
                className="item-header component-label"
                onMouseDown={[Function]}
                onMouseMove={[Function]}
                onMouseUp={[Function]}
              >
                X
              </div>
              <div
                className="item-content"
              >
                <input
                  className="input-component float-input"
                  onBlur={[Function]}
                  onChange={[Function]}
                  type="text"
                  value="1"
                />
              </div>
            </article>
            <article
              className="inspector-item wonder-float-input"
            >
              <div
                className="item-header component-label"
                onMouseDown={[Function]}
                onMouseMove={[Function]}
                onMouseUp={[Function]}
              >
                Y
              </div>
              <div
                className="item-content"
              >
                <input
                  className="input-component float-input"
                  onBlur={[Function]}
                  onChange={[Function]}
                  type="text"
                  value="1"
                />
              </div>
            </article>
            <article
              className="inspector-item wonder-float-input"
            >
              <div
                className="item-header component-label"
                onMouseDown={[Function]}
                onMouseMove={[Function]}
                onMouseUp={[Function]}
              >
                Z
              </div>
              <div
                className="item-content"
              >
                <input
                  className="input-component float-input"
                  onBlur={[Function]}
                  onChange={[Function]}
                  type="text"
                  value="1"
                />
              </div>
            </article>
          </div>
        </article>
      </article>
    </article>
    <article
      className="componentBox-component"
    >
      <div
        className="header"
      >
        <div
          className="header-triangle"
          onClick={[Function]}
        >
          <span
            className="triangle-bottom"
          />
        </div>
        <div
          className="header-title"
        >
          RenderGroup
        </div>
        <div
          className="header-close"
        >
          <img
            onClick={[Function]}
            src="./public/img/close.png"
          />
        </div>
      </div>
      <article
        className="wonder-render-group"
      >
        <div
          className="inspector-component"
        >
          <div
            className="component-title"
          >
            MeshRender
          </div>
          <hr />
          <div
            className="component-content"
          >
            <article
              className="wonder-mesh-renderer"
            >
              <article
                className="inspector-item"
              >
                <div
                  className="item-header"
                >
                  <span
                    className=""
                  >
                    Draw mode
                  </span>
                </div>
                <div
                  className="item-content"
                >
                  <select
                    onChange={[Function]}
                    value="4"
                  >
                    <option
                      value="0"
                    >
                      points
                    </option>
                    <option
                      value="1"
                    >
                      lines
                    </option>
                    <option
                      value="2"
                    >
                      line_loop
                    </option>
                    <option
                      value="3"
                    >
                      Line_strip
                    </option>
                    <option
                      value="4"
                    >
                      Triangles
                    </option>
                    <option
                      value="5"
                    >
                      triangle_strip
                    </option>
                    <option
                      value="6"
                    >
                      Triangle_fan
                    </option>
                  </select>
                </div>
              </article>
            </article>
          </div>
        </div>
        <div
          className="inspector-component"
        >
          <div
            className="component-title"
          >
            Material
          </div>
          <hr />
          <div
            className="component-content"
          >
            <article
              className="wonder-inspector-material"
            >
              <div
                className="inspector-item"
              >
                <div
                  className="item-header"
                >
                  Material
                </div>
                <div
                  className="item-content"
                >
                  <div
                    className="inspector-select"
                  >
                    <div
                      className="select-name"
                      onClick={[Function]}
                    >
                      Texture
                    </div>
                    <div
                      className="select-img"
                      onClick={[Function]}
                    >
                      <img
                        src="./public/img/select.png"
                      />
                    </div>
                  </div>
                </div>
              </div>
              <div
                className="material-value"
              >
                <article
                  className="inspector-item"
                >
                  <div
                    className="item-header"
                  >
                    <span
                      className=""
                    >
                      Type
                    </span>
                  </div>
                  <div
                    className="item-content"
                  >
                    <select
                      onChange={[Function]}
                      value="1"
                    >
                      <option
                        value="0"
                      >
                        basic_material
                      </option>
                      <option
                        value="1"
                      >
                        light_material
                      </option>
                    </select>
                  </div>
                </article>
                <div
                  className=""
                >
                  <article
                    className="wonder-light-material"
                  >
                    <article
                      className="inspector-item"
                    >
                      <div
                        className="item-header"
                      >
                        Diffcuse color
                      </div>
                      <div
                        className="item-content item-color"
                      >
                        <div
                          className="color-hex"
                          onClick={[Function]}
                          style={
                            Object {
                              "background": "#ffffff",
                            }
                          }
                        />
                        <div
                          className="color-select"
                          onClick={[Function]}
                        >
                          <img
                            src="./public/img/color.png"
                          />
                        </div>
                      </div>
                    </article>
                    <article
                      className="inspector-item"
                    >
                      <div
                        className="item-header"
                      >
                        Diffuse map
                      </div>
                      <div
                        className="item-content item-texture"
                      >
                        <div
                          className="texture-img"
                          onClick={[Function]}
                          style={
                            Object {
                              "opacity": "1",
                            }
                          }
                        >
                          <div
                            className="img-dragBg"
                            onDragEnter={[Function]}
                            onDragLeave={[Function]}
                            onDragOver={[Function]}
                            onDrop={[Function]}
                          />
                          <img
                            src="object_url0"
                          />
                        </div>
                        <button
                          className="texture-remove"
                          onClick={[Function]}
                        >
                          Remove
                        </button>
                      </div>
                    </article>
                    <article
                      className="inspector-item wonder-float-input"
                    >
                      <div
                        className="item-header component-label"
                        onMouseDown={[Function]}
                        onMouseMove={[Function]}
                        onMouseUp={[Function]}
                      >
                        Shininess
                      </div>
                      <div
                        className="item-content"
                      >
                        <input
                          className="input-component float-input"
                          onBlur={[Function]}
                          onChange={[Function]}
                          type="text"
                          value="32"
                        />
                      </div>
                    </article>
                  </article>
                </div>
              </div>
            </article>
          </div>
        </div>
      </article>
    </article>
    <article
      className="componentBox-component"
    >
      <div
        className="header"
      >
        <div
          className="header-triangle"
          onClick={[Function]}
        >
          <span
            className="triangle-bottom"
          />
        </div>
        <div
          className="header-title"
        >
          Geometry
        </div>
        <div
          className="header-close"
        >
          <img
            onClick={[Function]}
            src="./public/img/close.png"
          />
        </div>
      </div>
      <article
        className="wonder-inspector-geometry"
      >
        <div
          className="inspector-item"
        >
          <div
            className="item-header"
          >
            Geometry
          </div>
          <div
            className="item-content"
          >
            <div
              className="inspector-select"
            >
              <div
                className="select-name"
                onClick={[Function]}
              >
                Mesh
              </div>
              <div
                className="select-img"
                onClick={[Function]}
              >
                <img
                  src="./public/img/select.png"
                />
              </div>
            </div>
          </div>
        </div>
      </article>
    </article>
    <article
      className="wonder-addable-component"
    >
      <div
        className="addable-component-content"
      >
        <div
          className="content-btn"
        >
          <button
            className="addable-btn"
            onClick={[Function]}
          >
            Add Component
          </button>
        </div>
      </div>
    </article>
  </article>
</article>
`;

exports[`test MainEditorMaterialMap test select texture group -> show order 
        order should be:
        sort texture assets by firstname alphabetically
         1`] = `
<article
  className="inspector-item"
>
  <div
    className="item-header"
  >
    Diffuse map
  </div>
  <div
    className="item-content item-texture"
  >
    <div
      className="texture-img"
      onClick={[Function]}
      style={
        Object {
          "opacity": "1",
        }
      }
    >
      <div
        className="img-dragBg"
        onDragEnter={[Function]}
        onDragLeave={[Function]}
        onDragOver={[Function]}
        onDrop={[Function]}
      />
    </div>
    <button
      className="texture-remove"
      onClick={[Function]}
    >
      Remove
    </button>
  </div>
  <div
    className="select-component-content"
  >
    <div
      className="select-component-item"
    >
      <div
        className="select-item-header"
      >
        Texture
      </div>
      <div
        className="select-item-imgBody"
      >
        <div
          className="imgBody-content"
        >
          <div
            className="select-item-imgContent"
            onClick={[Function]}
          >
            <img
              className="imgContent-img"
              src="texture2img"
            />
            <div
              className="imgContent-text"
            >
              ATexture
            </div>
          </div>
          <div
            className="select-item-imgContent"
            onClick={[Function]}
          >
            <img
              className="imgContent-img"
              src="texture1img"
            />
            <div
              className="imgContent-text"
            >
              BTexture
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="select-component-bg"
      onClick={[Function]}
    />
  </div>
</article>
`;
